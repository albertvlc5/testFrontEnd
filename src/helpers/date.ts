export const getPreviousMonthNameByLocale = (
  dateStr: string,
  locale = 'pt-BR',
) => {
  const date = new Date(dateStr);
  const month = date.getUTCMonth();
  const previousMonth = month === 0 ? 11 : month - 1;

  const monthName = new Date(
    date.getUTCFullYear(),
    previousMonth,
    1,
  ).toLocaleString(locale, { month: 'long' });

  return monthName;
};
