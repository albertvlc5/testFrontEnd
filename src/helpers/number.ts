import 'intl';
import 'intl/locale-data/jsonp/pt';
export const formatNumberByLocale = (value: number, locale = 'pt-Br'): string =>
  new Intl.NumberFormat(locale, { minimumFractionDigits: 2 }).format(value);

export const formatMoney2Numeral = (money: string): number => {
  const [value, cents] = String(money).split(',');
  return Number([value.replace(/\D+/g, ''), cents].join('.'));
};

export const numberToCurrencyIntl = (
  value: number = 0,
  locale = 'pt-BR',
  currency = 'BRL',
) =>
  new Intl.NumberFormat(locale, { style: 'currency', currency }).format(value);
