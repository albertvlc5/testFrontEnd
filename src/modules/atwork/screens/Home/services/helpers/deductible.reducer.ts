import {
  IDeductible,
  IDeductibleTransformed,
} from '../../interfaces/deductible.interface';

export const initialDeductibleTransformed = {
  totalDiscounts: 0,
  dueDate: '',
};

export const deductibleReducer = (
  acumulator: IDeductibleTransformed,
  item: IDeductible,
) => ({
  totalDiscounts: acumulator.totalDiscounts + item.amountDeductibleThisMonth,
  dueDate: item.dueDate,
});
