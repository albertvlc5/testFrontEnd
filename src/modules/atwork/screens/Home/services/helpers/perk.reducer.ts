import { ProductType } from '../../interfaces/product-type.enum';
import { IPerk, IPerkTransformed } from '../../interfaces/perk.interface';

export const initialPerkTransformed = {
  totalAmount: 0,
  salaryAdvancement: {
    totalAmountAvailable: 0,
    amountDeductible: 0,
    currency: '',
  },
  payroll: {
    amountDeductible: undefined,
    totalAmountAvailable: 0,
    currency: '',
  },
};

export const perkReducer = (acumulator: IPerkTransformed, item: IPerk) => {
  const salaryAdvancement =
    item.type === ProductType.SALARY_ADVANCEMENT
      ? {
          totalAmountAvailable: item.totalAmountAvailable,
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
        }
      : acumulator.salaryAdvancement;

  const payroll =
    item.type === ProductType.PAYROLL
      ? {
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
          totalAmountAvailable: item.totalAmountAvailable,
        }
      : acumulator.payroll;

  return {
    totalAmount: acumulator.totalAmount + item.totalAmountAvailable,
    salaryAdvancement: salaryAdvancement,
    payroll: payroll,
  };
};
