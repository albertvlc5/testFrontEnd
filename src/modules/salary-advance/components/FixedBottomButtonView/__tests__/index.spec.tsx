import { fireEvent } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';
import { mountWithAtworkHost } from '../../../../../../src/__tests__/helpers';

import { FixedBottomButtonView } from '../index';

describe('Fixed Bottom Button View', () => {
  it('should render component', () => {
    const mockOnPress = jest.fn();
    const wrapper = mountWithAtworkHost(
      <FixedBottomButtonView onPress={mockOnPress} children={<View />} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should output onPress event when the button has been pressed', () => {
    const mockOnPress = jest.fn();
    const { getByTestId } = mountWithAtworkHost(
      <FixedBottomButtonView onPress={mockOnPress} children={<View />} />,
    );
    const comp = getByTestId('flat-button-id');
    fireEvent(comp, 'onPress');
    expect(mockOnPress).toHaveBeenCalled();
  });
});
