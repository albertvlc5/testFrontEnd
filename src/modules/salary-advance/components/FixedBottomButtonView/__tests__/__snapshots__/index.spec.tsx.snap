// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed Bottom Button View should render component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "height": 90,
          "position": "absolute",
          "right": 30,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#49E295",
              "borderRadius": 16,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            },
          ]
        }
        testID="flat-button-id"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={32}
          bbWidth={32}
          fill="none"
          focusable={false}
          height={32}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 32,
                "width": 32,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={32}
        >
          <RNSVGGroup
            fill={null}
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M19 12H5"
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "stroke",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={
                Array [
                  0,
                  4281549892,
                ]
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
            <RNSVGPath
              d="M14 17L19 12"
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "stroke",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={
                Array [
                  0,
                  4281549892,
                ]
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
            <RNSVGPath
              d="M14 7L19 12"
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "stroke",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={
                Array [
                  0,
                  4281549892,
                ]
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
